src/cmd/services/m3coordinator/downsample/downsampler.go

```diff
@@ -39,6 +39,7 @@ type MetricsAppender interface {
 	SamplesAppender(opts SampleAppenderOptions) (SamplesAppender, error)
 	Reset()
 	Finalize()
+	IsDropPolicyApplied() bool
```

Comment:
Hm, so since this is only active after calling `SamplesAppender(..)` - maybe we instead make this method return both the samples appender and the match result so it's clear it comes from that method?

i.e.
```go
type MetricsAppender interface {
  SamplesAppender() (SamplesAppenderResult, error)
}

type SamplesAppenderResult struct {
  SamplesAppender SamplesAppender
  IsDropPolicyApplied bool
}
```